File: Caper/CaperSharp/SequenceEngine.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CaperSharp
{
  public abstract class SequenceEngine
  {
    public string Path { get; set; }

    public Sequences Sequences { get; private set; }

    public SequenceEngine( string aPath )
    {
      Path = aPath;
      Sequences = new Sequences();
    }

    internal abstract bool Initialize();
  }
}

File: Caper/Caper/SequenceEngine.cpp
#include "SequenceEngine.h"

SequenceEngine::SequenceEngine(string aPath)
{
  mPath = aPath;
  mSequences = new Sequences();
  mContigs = new vector<string>();
}

SequenceEngine::~SequenceEngine()
{
  delete mSequences;
  delete mContigs;
}
